WordPress/UITestsFoundation/Screens/Editor/BlockEditorScreen.swift

```diff
@@ -234,37 +244,43 @@ public class BlockEditorScreen: ScreenObject {
         }
     }
 
-    public func publish() throws -> EditorNoticeComponent {
-        return try post(action: "Publish")
+    private enum postAction: String {
+        case publish = "Publish"
+        case schedule = "Schedule"
+    }
+
+    public func publish() throws {
+        return try post(action: .publish)
+    }
+
+    public func publishAndViewEpilogue() throws -> EditorPublishEpilogueScreen {
+        try publish()
+        waitAndTap(noticeViewButton)
+        return try EditorPublishEpilogueScreen()
     }
 
-    public func schedulePost() throws -> EditorNoticeComponent {
-        return try post(action: "Schedule")
+    public func schedulePost() throws {
+        try post(action: .schedule)
     }
 
-    private func post(action: String) throws -> EditorNoticeComponent {
-        let postButton = app.buttons[action]
-        let postNowButton = app.buttons["\(action) Now"]
+    public func schedulePostAndViewEpilogue() throws -> EditorPublishEpilogueScreen {
+        try schedulePost()
+        waitAndTap(noticeViewButton)
+        return try EditorPublishEpilogueScreen()
+    }
```

Comment:
i think we can DRY this further, possibly by moving the `ViewEpilogue` part of the method into the `post(action: postAction)`?  this way we won't need to create `publishAndViewEpilogue()` and `schedulePostAndViewEpilogue()`. wdyt?

something like:
```
    public func publish() throws -> EditorPublishEpilogueScreen {
        return try postAndViewEpilogue(action: .publish)
    }

    public func schedulePost() throws -> EditorPublishEpilogueScreen {
        return try postAndViewEpilogue(action: .schedule)
    }

    private func postAndViewEpilogue(action: postAction) throws -> EditorPublishEpilogueScreen {
        let postButton = app.buttons[action.rawValue]
        let postNowButton = app.buttons["\(action.rawValue) Now"]
        var tries = 0
        // This loop to check for Publish/Schedule Now Button is an attempt to confirm that the postButton.tap() call took effect.
        // The tests would fail sometimes in the pipeline with no apparent reason.
        repeat {
            postButton.tap()
            tries += 1
        } while !postNowButton.waitForIsHittable(timeout: 3) && tries <= 3

        try confirmPost(button: postNowButton, action: action)

        waitAndTap(noticeViewButton)
        return try EditorPublishEpilogueScreen()
    }
```

or if we want to keep post and view epilogue separate, we can DRY it by calling `post(action: postAction)` directly in `publishAndViewEpilogue()`:
```
    public func publishAndViewEpilogue() throws -> EditorPublishEpilogueScreen {
        try post(action: .publish)
        waitAndTap(noticeViewButton)
        return try EditorPublishEpilogueScreen()
    }
```

Reply:
yeah think that makes sense and very possible that we'd want to use it without viewing Epilogue so I think that's fine to keep it that way.

my second suggestion above is about `publishAndViewEpilogue()`, where instead of calling `publish()` or `schedulePost`, we call `try post(action: PostAction)` directly to remove that one extra level. not sure if it'll make a big difference in terms of timing but i think keeping the calls as direct as possible (when/if possible) usually helps with timing issues in UI tests. wdyt?

Reply:
Totally! Addressed in https://github.com/wordpress-mobile/WordPress-iOS/pull/21414/commits/0af55e837c5905c772c11a5c8631716a532bff5d.